### PR TITLE
Converter for LogicalNot operation

### PR DESCRIPTION
**Pull request by drivanov**
- Converter for LogicalNot operation is using the same base class (ConvertUnaryImpl) as other Unary operations.
- The special converter for Rsqrt was removed and now for this operation regular Unary Op converter is used.
- New templated class implemented for testing ConvertUnary, ConvertBooleanUnary, ConvertActivation.
- A new check and corresponding subtests were added to Validate: at least 1 dimension is required for input of any Unary and UnaryBoolean operation. (Similar subtest for ConvertActivation operations is blocked and after refactoring of ConvertActivation it will be activated)

### CODE REVIEW

**Comment by drivanov** [comment]
Replacement for [PR#55229](https://github.com/tensorflow/tensorflow/pull/55229)

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -1849,6 +1838,94 @@ class ParameterizedOpConverterTestBase
 };
 
 template <typename T>
+class OpConverter_UnaryTest : public ParameterizedOpConverterTestBase {
+ public:
+  template <typename S>
+  void RunTests(
+      const string& testName, std::vector<string>& ops_to_test,
+      const operationMap<S>& map,
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:1846]: Shouldn't the type be T?

**Review by bixia1 — COMMENTED** [review]
There are test build errors, I will attach a log.

**Comment by bixia1** [comment]
[fail1.log](https://github.com/tensorflow/tensorflow/files/8374305/fail1.log)

**Comment by drivanov** [comment]
> [fail1.log](https://github.com/tensorflow/tensorflow/files/8374305/fail1.log)
FIXED

Sorry, while resolving a merge conflict online, I didn't notice that the line `template <typename T>` disappeared from 
```
template <typename T>
class OpConverter_BinaryTest
```

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 1 above)
Comment (drivanov) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:1846]: No, we need both **T** and **S. T** is used for types (float, int,, bool etc) and **S** is used for enums (`nvinfer1::UnaryOperatrion, nvinfer1::ActivationType`) used in `operationMap<S>`

**Review by bixia1 — APPROVED** [review]

**Review by bixia1 — APPROVED** [review]

**Comment by bixia1** [comment]
@drivanov build errors:
In file included from [third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:16](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc?l=16&ws=tap-prod-presubmit/180112651&snapshot=2):
In file included from [./third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h:26](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h?l=26&ws=tap-prod-presubmit/180112651&snapshot=2):
In file included from [./third_party/tensorflow/compiler/tf2tensorrt/convert/op_converter.h:25](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/op_converter.h?l=25&ws=tap-prod-presubmit/180112651&snapshot=2):
In file included from [./third_party/tensorflow/compiler/tf2tensorrt/convert/weights.h:22](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/weights.h?l=22&ws=tap-prod-presubmit/180112651&snapshot=2):
[./third_party/tensorflow/compiler/tf2tensorrt/convert/utils.h:98](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/utils.h?l=98&ws=tap-prod-presubmit/180112651&snapshot=2):30: error: no matching function for call to 'DebugString'
    StrAppend(&tmp_s, StrCat(DebugString(el), ", "));
                             ^~~~~~~~~~~
[third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:1706](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc?l=1706&ws=tap-prod-presubmit/180112651&snapshot=2):58: note: in instantiation of function template specialization 'tensorflow::tensorrt::DebugString<bool>' requested here
                       << ", Received Input Tensor: " << DebugString(values);
                                                         ^
[third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:1888](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc?l=1888&ws=tap-prod-presubmit/180112651&snapshot=2):7: note: in instantiation of function template specialization 'tensorflow::tensorrt::convert::ParameterizedOpConverterTestBase::AddTestTensor<bool>' requested here
      AddTestTensor("input", p.input_dims, input_tf_type, input_values);
      ^
[third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:7057](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc?l=7057&ws=tap-prod-presubmit/180112651&snapshot=2):3: note: in instantiation of function template specialization 'tensorflow::tensorrt::convert::OpConverter_UnaryTest<bool>::RunTests<nvinfer1::UnaryOperation>' requested here
  RunTests("LogicalUnary", ops_to_test, *UnaryBooleanOperationMap(), op_map,
  ^
[./third_party/tensorflow/compiler/tf2tensorrt/convert/trt_parameters.h:46](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/trt_parameters.h?l=46&ws=tap-prod-presubmit/180112651&snapshot=2):8: note: candidate function not viable: no known conversion from 'const std::__bit_iterator<std::vector<bool>, true, 0>::reference' (aka 'const std::__bit_const_reference<std::vector<bool>>') to 'const tensorflow::tensorrt::TrtPrecisionMode' for 1st argument
string DebugString(const TrtPrecisionMode mode);
       ^
[./third_party/tensorflow/compiler/tf2tensorrt/convert/utils.h:95](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/utils.h?l=95&ws=tap-prod-presubmit/180112651&snapshot=2):8: note: candidate template ignored: could not match 'vector' against '__bit_const_reference'
string DebugString(const std::vector<CType>& vector) {
       ^
[./third_party/tensorflow/compiler/tf2tensorrt/convert/utils.h:86](https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/compiler/tf2tensorrt/convert/utils.h?l=86&ws=tap-prod-presubmit/180112651&snapshot=2):8: note: candidate template ignored: requirement 'std::is_arithmetic<std::__bit_const_reference<std::vector<bool, std::allocator<bool>>>>::value' was not satisfied [with CType = std::__bit_const_reference<std::vector<bool>>]
string DebugString(const CType& el) {

**Comment by drivanov** [comment]
@bixia1 : I don't see these compilation issues on my side. To make things easier, I just created [PR#55542](https://github.com/tensorflow/tensorflow/pull/55542) and am closing it.

**Pull request closed without merging**